### PR TITLE
Remove m5o code paths, fix some linter warnings

### PR DESCRIPTION
**Pull request by cjohnhanson**
Unsurprisingly, removing all of these code paths triggers a lot of flake8 warnings. Fixing all linter warnings would make for a pretty messy diff, so I'm opening this draft PR with just the functional code changes for review, and I'll start a separate branch off of this one to solve the linting issues and then open the final PR from that one after this has been approved. 

Local tests are passing and I did some manual smoke testing to confirm that these removed code paths don't break anything.

### CODE REVIEW

**Review by tayloramurphy — APPROVED** [review]

**Comment by tayloramurphy** [comment]
No docs changes I see, so approved from my end.

**Review by pandemicsyn — APPROVED** [review]

File: src/meltano/core/settings_service.py [hunk 1]
```diff
@@ -6,6 +6,8 @@
 from abc import ABC, abstractmethod
 from contextlib import contextmanager
 from enum import Enum
+from functools import reduce
+from operator import xor
```
Comment (edgarrmondragon) [inline src/meltano/core/settings_service.py:10]: probably a leftover?

**Review by edgarrmondragon — APPROVED** [review]

File: docs/src/_concepts/plugins.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 title: Plugins
 description: Meltano takes a modular approach to data engineering and EL(T), where your project and pipelines are composed of plugins.
 layout: doc
+redirect_from:
+  - /guide/analysis
```
Comment (cjohnhanson) [inline docs/src/_concepts/plugins.md:6]: @tayloramurphy Sure thing, here's the redirect.

File: docs/src/_reference/command-line-interface.md [hunk 3]
```diff
@@ -87,7 +87,7 @@ meltano add extractor tap-ga--client-foo --inherit-from tap-google-analytics
 
 - `--variant=<variant>`: Add a specific (non-default) [variant](/concepts/plugins#variants) of the identified [discoverable plugin](/concepts/plugins#discoverable-plugins).
 
-- `--include-related`: Also add transform, dashboard, and model plugins related to the identified discoverable extractor.
+- `--include-related`: Also add transform plugins related to the identified discoverable extractor.
```
Comment (aaronsteers) [inline docs/src/_reference/command-line-interface.md:91]: @edgarrmondragon - Does your latest update honor `--include-related` when considering `requires`? 

If not, I think we should just remove this:

```suggestion
```

If so, I would reword:

```suggestion
- `--include-related`: Adds related plugins if they are required and not yet added to the project.

```

File: docs/src/_reference/command-line-interface.md [hunk 4]
```diff
@@ -483,7 +481,7 @@ Installs dependencies of your project based on the **meltano.yml** file.
 Optionally, provide a plugin type argument to only (re)install plugins of a certain type.
 Additionally, plugin names can be provided to only (re)install those specific plugins.
 
-Use `--include-related` to automatically install transform, model, and dashboard plugins related to installed extractor plugins.
+Use `--include-related` to automatically install transforms related to installed extractor plugins.
```
Comment (aaronsteers) [inline docs/src/_reference/command-line-interface.md:485]: Ditto above. I think we delete:

```suggestion
```

or reword:

```suggestion
Use `--include-related` to automatically install related plugins in required and not yet added to the project.

```

**Review by aaronsteers — APPROVED** [review]
@edgarrmondragon, @cjohnhanson, @kgpayne - We've got some `--add-related` refs that I think will be dropped. Realized a little late in my review that #5957 is a separate MR.

This looks good to go. My comments may be helpful for #5957 but shouldn't need to be resolved here.

**Comment by aaronsteers** [comment]
@cjohnhanson - is this okay to merge? I see lint failure, which we agreed is okay at this point.
